flexget/plugins/sites/newznab.py

```diff
@@ -1,125 +1,615 @@
+        except RequestException as ex:
+            raise plugin.PluginError("Failed fetching url: %s error: %s" % (url, ex))
+
+        if r and r.status_code != 200:
+            raise plugin.PluginError('Unable to reach indexer url: %s' % url)
+
+        xml_data = r.content
+        try:
+            xml_data = self._make_feedparser_friendly(xml_data)
+        except Exception as ex:
+            log.debug('Could not apply feedparser fix, trying without. Error: %s', ex)
+
+        try:
+            parsed_xml = feedparser.parse(xml_data)
+        except Exception as ex:
+            raise plugin.PluginError('Unable to parse XML from url: %s error: %s' % (url, ex))
+
+        if 'error' in parsed_xml.feed:
+            feed = parsed_xml.feed
+            if 'code' in feed['error']:
+                if 'description' in feed['error']:
+                    log.error('Error code: %s detail: %s', feed['error']['code'], feed['error']['description'])
+                else:
+                    log.error('Error code: %s', feed['error']['code'])
+            raise plugin.PluginError('Parsed XML is a error return for url: %s' % url)
+
+        if not parsed_xml.entries:
+            log.info('No entries returned from xml.')
             return []
-        if arg_entry.get('tvrage_id'):
-            lookup = '&rid=%s' % arg_entry.get('tvrage_id')
+
+        entries = self.parse_from_xml(parsed_xml.entries)
+        if not entries:
+            log.verbose('No entries parsed from xml.')
+
+        return entries
+
+    def _build_query_url_fragment(self, query_string, config):
+        query_string = query_string.strip()
+        if not query_string:
+            log.error('Invalid/empty query_string given: %s', query_string)
+            return None
+        query = normalize_unicode(query_string)
+        # query = normalize_scene(query)
+        query = quote_plus(query.encode('utf8'))
+        if config['force_quotes'] is True:
+            query = "&q=\"%s\"" % query
         else:
-            lookup = '&q=%s' % quote(arg_entry['series_name'])
-        url = config['url'] + lookup + '&season=%s&ep=%s' % (arg_entry['series_season'], arg_entry['series_episode'])
-        return self.fill_entries_for_url(url, task)
+            query = "&q=%s" % query
+        return query
 
-    def do_search_movie(self, arg_entry, task, config=None):
-        entries = []
-        log.info('Searching for %s (imdbid:%s)' % (arg_entry['title'], arg_entry['imdb_id']))
-        # normally this should be used with emit_movie_queue who has imdbid (i guess)
-        if 'imdb_id' not in arg_entry:
-            return entries
-
-        imdb_id = arg_entry['imdb_id'].replace('tt', '')
-        url = config['url'] + '&imdbid=' + imdb_id
-        return self.fill_entries_for_url(url, task)
+    def _build_metaid_url_fragment(self, entry):
+        url_param = None
+        # use first valid meta id
+        if self._is_tv_entry(entry) is True:
```

Comment:
If you wish to support entries that are neither movies nor series, then I'd rather you made it more explicit and had separate boolean functions for each. You have to consider that other people will be reading your code eventually and if a boolean function has a third special return value, then it becomes more difficult to read.